### ISSUE TITLE
Is is possible to use column searching with datatables service?

### ISSUE THREAD

**Issue by gn0rt0n** [yajra/laravel-datatables#387]
If this is possible, what would the syntax be? The following does not work, but I'm not sure if it can even be done.

```
 public function html()
    {
        return $this->builder()
            ->columns($this->getColumns())
            ->ajax('')
            ->addAction(['width' => '80px'])
            ->parameters($this->getBuilderParameters())
            ->initComplete(function(){
                this.api().columns().every(function(){
                    var column = this;
                    var input = document.createElement("input");
                    $(input).appendTo($(column.footer()).empty())
                    .on('change', function () {
                        column.search($(this).val(), false, false, true).draw();
                    });
                });
            });
    }
```

**Comment by yajra** [comment]
Unfortunately, `initComplete` and most parameters that requires a function is not yet supported by the builder. However, we can do a workaround here by hooking on [DT Events](https://datatables.net/reference/event/) and I think `init.dt` event would help you achieve what you want to do?

**Comment by gn0rt0n** [comment]
I think this might work. However, I have a problem with the datatable not having a tfoot row to manipulate. The datable is generated with the "no-footer" class, so it seems that somewhere it is explicitly being told not to generate the footer, but I can't tell where. I have tried to track it down, but haven't had any luck yet. I could mess with the dom itself and just create one, but that seems sloppy since datatables is supposed to support it.

If you have any information on why the tfoot would not be generated I would appreciate it. It seems that since there is no footer element to manipulate, the code I am using exits without returning results. 

I have the following:
```
 $('#dataTableBuilder').on('init.dt', function () {

            var table = $('#dataTableBuilder');
            var api = table.dataTable().api();
            api.columns().every(function () {
                var column = this;
                var input = document.createElement("input");
                $(input).appendTo($(column.footer()).empty())
                    .on('change', function () {
                        column.search($(this).val(), false, false, true).draw();
                    });
            });
        })
```

**Comment by yajra** [comment]
@gn0rt0n, if you need a footer. I think you can just manually write the table instead of using `$dataTable->table()` helper? Just remember to put an `id="dataTableBuilder"`.

**Comment by yajra** [comment]
I just released v6.5.0 which will support DT callbacks. See #401 for example usage. :beers:
